client/clip_client/client.py

```diff
@@ -159,17 +159,30 @@ def _iter_doc(self, content) -> Generator['Document', None, None]:
             if isinstance(c, str):
                 _mime = mimetypes.guess_type(c)[0]
                 if _mime and _mime.startswith('image'):
-                    yield Document(
-                        tags={'__created_by_CAS__': True}, uri=c
-                    ).load_uri_to_blob()
+                    try:
+                        yield Document(
+                            tags={'__created_by_CAS__': True}, uri=c
+                        ).load_uri_to_blob()
+                    except Exception:
+                        raise ValueError(
+                            f'Failed to load image {c}. Maybe the URI is broken?'
+                        )
```

Comment:
why raise a custom exception? why not throw the exception from docarray?

Reply:
It's kinda too long?

Reply:
it's not about the length. It's about the information we can provide to user.

Reply:
If the exception from docarray is not good enough, then improve the docarray.